src/contour/Config.h

```diff
@@ -144,138 +150,332 @@ struct SimpleColorConfig
+        }
+
+        TerminalProfile const* profile(std::string const& name) const
+        {
+            assert(!name.empty());
+            if (auto i = profiles.find(name); i != profiles.end())
+                return &i->second;
+            assert(false && "Profile not found.");
+            return nullptr;
+        }
+
+        TerminalProfile& profile() noexcept
+        {
+            if (auto* prof = profile(defaultProfileName); prof)
+                return *prof;
+            crispy::unreachable();
+        }
+        TerminalProfile const& profile() const noexcept
+        {
+            if (auto const* prof = profile(defaultProfileName); prof)
+                return *prof;
+            crispy::unreachable();
+        }
+
+        // selection
+        std::string wordDelimiters;
+        vtbackend::Modifier bypassMouseProtocolModifier = vtbackend::Modifier::Shift;
+        SelectionAction onMouseSelection = SelectionAction::CopyToSelectionClipboard;
+        vtbackend::Modifier mouseBlockSelectionModifier = vtbackend::Modifier::Control;
+
+        // input mapping
+        InputMappings inputMappings;
+
+        bool spawnNewProcess = false;
+        std::shared_ptr<logstore::sink> loggingSink;
+
+        bool sixelScrolling = true;
+        vtbackend::ImageSize maxImageSize = {}; // default to runtime system screen size.
+        unsigned maxImageColorRegisters = 4096;
+
+        std::set<std::string> experimentalFeatures;
+    };
+} // namespace old
+
+template <typename T>
+struct ConfigEntry
 {
-    std::filesystem::path backingFilePath;
+    using val_type = T;
 
-    bool live = false;
+    constexpr ConfigEntry(std::string d): _val {}, _documentation { std::move(d) } {}
 
-    /// Qt platform plugin to be loaded.
-    /// This is equivalent to QT_QPA_PLATFORM.
-    std::string platformPlugin;
+    constexpr ConfigEntry(T&& in, std::string d):
+        _val { std::forward<T>(in) }, _documentation { std::move(d) }
+    {
+    }
 
-    RenderingBackend renderingBackend = RenderingBackend::Default;
+    template <typename F>
+        requires std::is_constructible_v<T, F>
+    constexpr ConfigEntry(F in, std::string d): _val { static_cast<T>(in) }, _documentation { std::move(d) }
+    {
+    }
 
-    /// Enables/disables support for direct mapped texture atlas tiles (e.g. glyphs).
-    bool textureAtlasDirectMapping = true;
+    [[nodiscard]] constexpr T const& get() const { return _val; }
+    [[nodiscard]] constexpr T& get() { return _val; }
 
-    // Number of hashtable slots to map to the texture tiles.
-    // Larger values may increase performance, but too large may also decrease.
-    // This value is rounted up to a value equal to the power of two.
-    //
-    // Default: 4096
-    crispy::strong_hashtable_size textureAtlasHashtableSlots = crispy::strong_hashtable_size { 4096 };
+    [[nodiscard]] constexpr std::string makeDoc() const
+    {
+        return fmt::format(fmt::runtime(_documentation), _val);
```

Comment:
```suggestion
        return fmt::format(fmt::runtime(_documentation), _value);
```

nice, TIL (`fmt::runtime...` :) )